connectionmgr.go

```diff
@@ -0,0 +1,72 @@
+package utils
```

Comment:
Every consumer of utils doesn't need this file, it should be put into a separate package for this module. Suggest `connection_manager.go`